### PR TITLE
Split UnnecessaryApply tests

### PR DESCRIPTION
**Pull request by arturbosch**
- fixes formatting issues due to activation of detekt-formatting
- splits positive test cases

### CODE REVIEW

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryApplySpec.kt [hunk 1]
```diff
@@ -8,33 +8,60 @@ import org.jetbrains.spek.api.dsl.it
...
+
+		it("does report single statement in apply used as function argument") {
+			assertThat(subject.lint("""
+				fun b(i : Int?) {
+				}
+				fun f() {
+					val a : Int? = null
 					b(a.apply {
-						plus(1)
+						toString()
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryApplySpec.kt]: Actually this case is wrong, I have changed the test to use `toString()` in the apply which clearly leads to a type error for function `b`.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryApplySpec.kt [hunk 2]
```diff
@@ -8,33 +8,60 @@ import org.jetbrains.spek.api.dsl.it
...
 class UnnecessaryApplySpec : SubjectSpek<UnnecessaryApply>({
+
 	subject { UnnecessaryApply(Config.empty) }
 
-	given("some code using apply expressions extensively") {
-		it("reports unnecessary applies that can be changed to ordinary method call") {
-			val findings = subject.lint("""
-				fun b(i : Int) {
-				}
+	given("unnecessary applies that can be changed to ordinary method call") {
```
Comment (Mauin) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryApplySpec.kt]:
```suggestion
	given("unnecessary apply expressions that can be changed to ordinary method call") {
```

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryApplySpec.kt [hunk 3]
```diff
@@ -8,33 +8,60 @@ import org.jetbrains.spek.api.dsl.it
...
+
+		it("reports an apply on non-nullable type") {
```
Comment (Mauin) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryApplySpec.kt]: The test description says the type is non-nullable. But `a` in this test case is `Int?` and thus nullable.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryApplySpec.kt [hunk 4]
```diff
@@ -8,33 +8,60 @@ import org.jetbrains.spek.api.dsl.it
...
+			assertThat(subject.lint("""
 				fun f() {
 					val a : Int? = null
```
Comment (Mauin) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryApplySpec.kt]: I guess this should be non-nullable?

```suggestion
					val a : Int = null
```

**Pull request merged**